src/static_canvas.class.ts

```diff
@@ -9,1855 +8,1834 @@ import { requestAnimFrame } from './util/animate';
-          this.createSVGClipPathMarkup(options),
-          '</defs>\n'
-        );
-      },
-
-      createSVGClipPathMarkup: function (options) {
-        var clipPath = this.clipPath;
-        if (clipPath) {
-          clipPath.clipPathId = 'CLIPPATH_' + uid();
-          return (
-            '<clipPath id="' +
-            clipPath.clipPathId +
-            '" >\n' +
-            this.clipPath.toClipPathSVG(options.reviver) +
-            '</clipPath>\n'
-          );
-        }
-        return '';
-      },
-
-      /**
-       * Creates markup containing SVG referenced elements like patterns, gradients etc.
-       * @return {String}
-       */
-      createSVGRefElementsMarkup: function () {
-        var _this = this,
-          markup = ['background', 'overlay'].map(function (prop) {
-            var fill = _this[prop + 'Color'];
-            if (fill && fill.toLive) {
-              var shouldTransform = _this[prop + 'Vpt'],
-                vpt = _this.viewportTransform,
-                object = {
-                  width: _this.width / (shouldTransform ? vpt[0] : 1),
-                  height: _this.height / (shouldTransform ? vpt[3] : 1),
-                };
-              return fill.toSVG(object, {
-                additionalTransform: shouldTransform
-                  ? fabric.util.matrixToSVG(vpt)
-                  : '',
-              });
-            }
+    markup.push(
+      '<svg ',
+      'xmlns="http://www.w3.org/2000/svg" ',
+      'xmlns:xlink="http://www.w3.org/1999/xlink" ',
+      'version="1.1" ',
+      'width="',
+      width,
+      '" ',
+      'height="',
+      height,
+      '" ',
+      viewBox,
+      'xml:space="preserve">\n',
+      '<desc>Created with Fabric.js ',
+      VERSION,
+      '</desc>\n',
+      '<defs>\n',
+      this.createSVGFontFacesMarkup(),
+      this.createSVGRefElementsMarkup(),
+      this.createSVGClipPathMarkup(options),
+      '</defs>\n'
+    );
+  }
+
+  createSVGClipPathMarkup(options: TSVGExportOptions): string {
+    const clipPath = this.clipPath;
+    if (clipPath) {
+      clipPath.clipPathId = `CLIPPATH_${uid()}`;
+      return (
+        '<clipPath id="' +
+        clipPath.clipPathId +
+        '" >\n' +
+        this.clipPath.toClipPathSVG(options.reviver) +
+        '</clipPath>\n'
+      );
+    }
+    return '';
+  }
+
+  /**
+   * Creates markup containing SVG referenced elements like patterns, gradients etc.
+   * @return {String}
+   */
+  createSVGRefElementsMarkup(): string {
+    return ['background', 'overlay']
+      .map((prop) => {
+        const fill = this[prop + 'Color'];
+        if (isFiller(fill)) {
+          const shouldTransform = this[prop + 'Vpt'],
+            vpt = this.viewportTransform,
+            object = {
+              width: this.width / (shouldTransform ? vpt[0] : 1),
+              height: this.height / (shouldTransform ? vpt[3] : 1),
+            };
+          return fill.toSVG(object as Rect, {
```

Comment:
I don't like `object as Rect`
I prefer fixing the types in the Filler

Reply:
yeah so object needs to be a fabricObject, but it can be FabricObject | StaticCanvas to be honest, it  just needs to have TSize or extends the interface of a TSize

Reply:
So let's change the signature at `fill#toSVG`